validator/keymanager/imported/import.go

```diff
@@ -20,26 +20,61 @@ import (
 func (km *Keymanager) ImportKeystores(
 	ctx context.Context,
 	keystores []*keymanager.Keystore,
-	importsPassword string,
-) error {
+	passwords []string,
+) ([]*ethpbservice.ImportedKeystoreStatus, error) {
+
+	var singlePasswordForAll string
+	if len(passwords) == 0 {
+		return nil, errors.New("no passwords provided for keystores")
+	} else if len(passwords) == 1 {
+		singlePasswordForAll = passwords[0]
+	} else {
+		if len(passwords) != len(keystores) {
+			return nil, errors.New("number of passwords does not match number of keystores")
+		}
+	}
+
 	decryptor := keystorev4.New()
 	bar := initializeProgressBar(len(keystores), "Importing accounts...")
 	keys := map[string]string{}
+	statuses := make([]*ethpbservice.ImportedKeystoreStatus, len(keystores))
 	var err error
+
 	for i := 0; i < len(keystores); i++ {
 		var privKeyBytes []byte
 		var pubKeyBytes []byte
-		privKeyBytes, pubKeyBytes, importsPassword, err = km.attemptDecryptKeystore(decryptor, keystores[i], importsPassword)
+		var passwordToUse string
+		if singlePasswordForAll != "" {
+			passwordToUse = singlePasswordForAll
+		} else {
+			passwordToUse = passwords[i]
+		}
+		privKeyBytes, pubKeyBytes, _, err = km.attemptDecryptKeystore(decryptor, keystores[i], passwordToUse)
 		if err != nil {
-			return err
+			statuses[i] = &ethpbservice.ImportedKeystoreStatus{
+				Status:  ethpbservice.ImportedKeystoreStatus_ERROR,
+				Message: err.Error(),
+			}
+			continue
 		}
 		// if key exists prior to being added then output log that duplicate key was found
+		if err := bar.Add(1); err != nil {
+			statuses[i] = &ethpbservice.ImportedKeystoreStatus{
+				Status:  ethpbservice.ImportedKeystoreStatus_ERROR,
+				Message: err.Error(),
+			}
+			continue
```

Comment:
If the progress bar is broken, let's just continue. I don't see a relationship between the progress bar failing and the import failing.

Reply:
Agreed!